lib_eio/core/dla.ml

```diff
@@ -0,0 +1,34 @@
+module Domain_local_await = Kcas.Domain_local_await
+
+let prepare_for_await mode =
+  let state =
+    Atomic.make
+      (mode :> [ Domain_local_await.mode | `Released | `Awaiting of _ ])
+  in
+  let release () =
+    if Atomic.get state != `Released then
+      match Atomic.exchange state `Released with
+      | `Awaiting b -> Broadcast.resume_all b
+      | _ -> ()
+  and await () =
+    match Atomic.get state with
+    | `Released -> ()
+    | `Awaiting _ -> failwith "await called twice"
+    | (`Cancelable | `Protected) as was ->
+        let b = Broadcast.create () in
```

Comment:
`Broadcast` seems a bit heavy-weight for this. It allows any number of suspenders, but it looks like this code only ever has at most one (given the `failwith` above). Can't we just use `state` to hold the `enqueue` function?

Reply:
I changed the PR to use domain local await that will hopefully soon be available from opam and simplified the implementation to just use `enqueue`.